flaskbb/forum/models.py

```diff
@@ -207,8 +207,13 @@ def save(self, user=None, topic=None):
         # Adding a new post
         if user and topic:
             created = time_utcnow()
-            self.user = user
-            self.username = user.username
+            if isinstance(user, db.Model):
```

Comment:
I don't like this, type checking is usually frowned up and here it's not type checking accurately enough to be really be useful. Rather, if a check is going to be here it should `if user.authenticated()`